In [1]:
import pandas as pd
import numpy as np
import os

In [2]:
path = r'/Users/sebreawilmore/Instacart Basket Analysis'

In [3]:
df_prods = pd.read_csv(os.path.join(path, 'Data', 'products_cleaned.csv'), index_col = False)

In [4]:
df_ords = pd.read_csv(os.path.join(path, 'Data', 'orders_wrangled.csv'), index_col = False)

# 2. 

In [7]:
df_prods.describe()

Unnamed: 0    product_id      aisle_id  department_id        prices
count  49672.000000  49672.000000  49672.000000   49672.000000  49672.000000
mean   24852.005053  24850.349775     67.762442      11.728942      9.993282
std    14342.265579  14340.705287     38.315784       5.850779    453.615536
min        0.000000      1.000000      1.000000       1.000000      1.000000
25%    12432.750000  12432.750000     35.000000       7.000000      4.100000
50%    24851.500000  24850.500000     69.000000      13.000000      7.100000
75%    37272.250000  37268.250000    100.000000      17.000000     11.100000
max    49692.000000  49688.000000    134.000000      21.000000  99999.000000

The maximum value for 'prices' is significantly higher than the IQR values. 

# 3.

In [8]:
for col in df_ords.columns.tolist():
  weird = (df_ords[[col]].applymap(type) != df_ords[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df_ords[weird]) > 0:
    print (col)

# 4.

No mixed data types present

# 5. 

In [9]:
df_ords.isnull().sum()

Unnamed: 0                     0
order_id                       0
user_id                        0
order_number                   0
orders_day_of_week             0
order_hour_of_day              0
days_since_prior_order    206209
dtype: int64

In [10]:
df_nan_ords = df_ords[df_ords['days_since_prior_order'].isnull()==True]

In [11]:
df_nan_ords

Unnamed: 0  order_id  user_id  order_number  orders_day_of_week  \
0                 0   2539329        1             1                   2   
11               11   2168274        2             1                   2   
26               26   1374495        3             1                   1   
39               39   3343014        4             1                   6   
45               45   2717275        5             1                   3   
...             ...       ...      ...           ...                 ...   
3420930     3420930    969311   206205             1                   4   
3420934     3420934   3189322   206206             1                   3   
3421002     3421002   2166133   206207             1                   6   
3421019     3421019   2227043   206208             1                   1   
3421069     3421069   3154581   206209             1                   3   

         order_hour_of_day  days_since_prior_order  
0                        8                     NaN  
11                      11                     NaN  
26                      14                     NaN  
39                      11                     NaN  
45                      12                     NaN  
...                    ...                     ...  
3420930                 12                     NaN  
3420934                 18                     NaN  
3421002                 19                     NaN  
3421019                 15                     NaN  
3421069                 11                     NaN  

[206209 rows x 7 columns]

The missing values may be attributed to customers being first time users of the Instacart app. Therefore, there will be no value indicating the days since their last order has been placed. The max value (206209) for the 'user_id' field corresponds with the number of missing values for the 'days_since_prior order' (206209) thus signifying one record for each new customer 

# 6.

In [ ]:
I have decided to leave the missing values as is. Imputing values is not necessary for the objective of this analysis,
and the statistical measures can still be analyzed with the missing values present

# 7.

In [20]:
df_ords = pd.read_csv(os.path.join(path, 'Data', 'orders_wrangled.csv'), index_col = False)

In [22]:
df_ords.shape

(3421083, 7)

In [25]:
df_ords_no_dups = df_ords.drop_duplicates()

In [26]:
df_ords_no_dups.shape

(3421083, 7)

No duplicates found

# 8.

No changes will be made to the data